### PR TITLE
perf(arm support): bot support deploy to customized resource group

### PR DESCRIPTION
**Pull request by dilin-MS**
bot bicep files returns bot web app resource id. During deploy project step, bot plugin get subscription id and resource group id from bot web app resource id. So we can deploy bot azure resources to customized resource group.

Task Item
- [[bot] bicep output resource id](https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/10793666)
- [[bot] deploy use bot resource id instead of resource group + sitename](https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/10793667)

### CODE REVIEW

File: packages/fx-core/src/plugins/resource/bot/plugin.ts [hunk 1]
```diff
@@ -540,28 +575,21 @@ export class TeamsBotImpl {
...
     await LanguageStrategy.localBuild(
       this.config.scaffold.programmingLanguage!,
-      packDir,
+      workingDir,
       this.config.deploy.unPackFlag === "true" ? true : false
     );
 
     await handler?.next(ProgressBarConstants.DEPLOY_STEP_ZIP_FOLDER);
-    const zipBuffer = utils.zipAFolder(packDir, DeployConfigs.UN_PACK_DIRS, [
+    const zipBuffer = utils.zipAFolder(workingDir, DeployConfigs.UN_PACK_DIRS, [
```
Comment (blackchoey) [inline packages/fx-core/src/plugins/resource/bot/plugin.ts:586]: it seems there're some unrelated changes, is it caused by git merge?

File: packages/fx-core/src/plugins/resource/bot/plugin.ts [hunk 2]
```diff
@@ -517,17 +543,26 @@ export class TeamsBotImpl {
...
     this.ctx = context;
     await this.config.restoreConfigFromContext(context);
+
+    this.config.provision.subscriptionId = getSubscriptionIdFromResourceId(
+      this.config.provision.botWebAppResourceId!
+    );
+    this.config.provision.resourceGroup = getResourceGroupNameFromResourceId(
+      this.config.provision.botWebAppResourceId!
+    );
+
```
Comment (blackchoey) [inline packages/fx-core/src/plugins/resource/bot/plugin.ts:553]: we also need to extract the web app name from resource id?

File: packages/fx-core/src/plugins/resource/bot/plugin.ts (hunk 1 above)
Comment (dilin-MS) [reply packages/fx-core/src/plugins/resource/bot/plugin.ts:586]: No. It is just some tiny code refactoring and I change it in passing.

Comment (dilin-MS) [reply packages/fx-core/src/plugins/resource/bot/plugin.ts:586]: The original `packDir` is just the same variable as the `workingDir`. They are just the same value and change the name. So I remove one of them.

File: packages/fx-core/src/plugins/resource/bot/plugin.ts (hunk 2 above)
Comment (dilin-MS) [reply packages/fx-core/src/plugins/resource/bot/plugin.ts:559]: Yes. Fixed.

**Review by blackchoey — APPROVED** [review]

**Review by YitongFeng-git — APPROVED** [review]
